sdk/src/main/scala/com/horizen/state/BaseStateReader.scala

```diff
@@ -2,12 +2,9 @@ package com.horizen.state
 
 import com.horizen.account.state.WithdrawalRequest
 import com.horizen.block.WithdrawalEpochCertificate
-import com.horizen.consensus.ConsensusEpochNumber
-import com.horizen.utils.{BlockFeeInfo, WithdrawalEpochInfo}
+import com.horizen.utils.BlockFeeInfo
 
 trait BaseStateReader {
-  def getWithdrawalEpochInfo: WithdrawalEpochInfo
```

Comment:
Note: here we actually left available for the MsgProcessors to get access to all other methods of the `BaseStateReader` for `BaseAccountStateView`, but from other side removed an access to the `getWithdrawalEpochInfo` and `getConsensusEpochNumber` from `AccountStateView`.
Let's keep this class as is.
But change the overall hierarchy of thew classes:
It was `BaseStateReader -> AccountStateReader -> BaseAccountStateView -> AccountStateView`
can be
```
BaseAccountStateReader -> BaseAccountStateView (empty)
                               \                                                                      
BaseStateReader ------------>  AccountStateReader   -> AccountStateView
```
`BaseAccountStateReader` contains the actual `AccountStateReader` methods but doesn't extend `BaseStateReader`.
Moreover `BaseAccountStateView` seems to be empty now. So we even can remove it and use `BaseAccountStateReader`

Reply:
I don't understand what you want to achieve yet. The way I see it we currently have three traits roughly corresponding to this:
- `BaseStateReader` base sidechain stuff, consensus epoch, withdrawal epoch, etc.
- `AccountStateReader` read-only account state access, like getBalance, getNonce, etc.
- `BaseAccountStateView` write access to state, like addBalance, updateAccountStorage, etc.

These traits extend each other like you said:
`BaseStateReader -> AccountStateReader -> BaseAccountStateView`

Usages are:
- `AccountState` implements `AccountStateReader` (and thus `BaseStateReader`)
- `AccountStateView` implements `BaseAccountStateView` and therefore all three
- `MessageProcessor`s  have access to `BaseAccountStateView`

Do you want to prevent `MessageProcessors` from having access to `BaseStateReader`?

Reply:
right, `MessageProcessors` should not be able to use/see `BaseStateReader` methods. In case using of the StateView in the past -> the methods output may be irrelevant.